### ISSUE TITLE
error: unexpected error on the SQL API side, please check your deployment logs

### ISSUE THREAD

**Issue by archiewood** [cube-js/cube.js#5095]
**Version:**
<img width="501" alt="image" src="https://user-images.githubusercontent.com/58074498/184212683-b961bc4a-440f-494b-974c-e0cd07e19b0f.png">

**Additional context**
- I am trying to use [Evidence](evidence.dev/), a BI tool to connect to Cube.js. 
- Under the hood, evidence uses the npm pg package (https://www.npmjs.com/package/pg)
- **It generally works (which is great)**
- However, when an incorrect query is entered, it crashes - full trace below

![24d3-afcf-8c0c-aacd9eef92d7](https://user-images.githubusercontent.com/58074498/184211539-63c92ef6-cbc8-4ea1-9c14-01eb09ac4651.gif)

I have tested Evidence with another postgres database, and incorrect queries do not crash, they just return a query error.

This makes me suspect it is **either** Cube.js **or** the npm `pg` package that has some kind of issue.

```SQLCompilationError: Internal: Initial planning error: Error during planning: Table or CTE with name 'order' not found QUERY: SELECT * FROM order LIMIT 10
node:events:491
      throw er; // Unhandled 'error' event
      ^

error: unexpected error on the SQL API side, please check your deployment logs
    at Parser.parseErrorMessage (/Users/archie/Projects/evidence2/node_modules/.pnpm/pg-protocol@1.5.0/node_modules/pg-protocol/dist/parser.js:287:98)
    at Parser.handlePacket (/Users/archie/Projects/evidence2/node_modules/.pnpm/pg-protocol@1.5.0/node_modules/pg-protocol/dist/parser.js:126:29)
    at Parser.parse (/Users/archie/Projects/evidence2/node_modules/.pnpm/pg-protocol@1.5.0/node_modules/pg-protocol/dist/parser.js:39:38)
    at Socket.<anonymous> (/Users/archie/Projects/evidence2/node_modules/.pnpm/pg-protocol@1.5.0/node_modules/pg-protocol/dist/index.js:11:42)
    at Socket.emit (node:events:513:28)
    at Socket.emit (node:domain:489:12)
    at addChunk (node:internal/streams/readable:324:12)
    at readableAddChunk (node:internal/streams/readable:297:9)
    at Readable.push (node:internal/streams/readable:234:10)
    at TCP.onStreamRead (node:internal/stream_base_commons:190:23)
Emitted 'error' event on BoundPool instance at:
    at Client.idleListener (/Users/archie/Projects/evidence2/node_modules/.pnpm/pg-pool@3.4.1_pg@8.7.1/node_modules/pg-pool/index.js:57:10)
    at Client.emit (node:events:513:28)
    at Client.emit (node:domain:489:12)
    at Client._handleErrorEvent (/Users/archie/Projects/evidence2/node_modules/.pnpm/pg@8.7.1/node_modules/pg/lib/client.js:319:10)
    at Client._handleErrorMessage (/Users/archie/Projects/evidence2/node_modules/.pnpm/pg@8.7.1/node_modules/pg/lib/client.js:330:12)
    at Connection.emit (node:events:513:28)
    at Connection.emit (node:domain:489:12)
    at /Users/archie/Projects/evidence2/node_modules/.pnpm/pg@8.7.1/node_modules/pg/lib/connection.js:114:12
    at Parser.parse (/Users/archie/Projects/evidence2/node_modules/.pnpm/pg-protocol@1.5.0/node_modules/pg-protocol/dist/parser.js:40:17)
    at Socket.<anonymous> (/Users/archie/Projects/evidence2/node_modules/.pnpm/pg-protocol@1.5.0/node_modules/pg-protocol/dist/index.js:11:42) {
  length: 99,
  severity: 'FATAL',
  code: 'XX000',
  detail: undefined,

...(and more stuff about the connection, such as my db credentials)
```

**Comment by ovr** [comment]
Hello @archiewood ,

Could you please try with latest version?

Thanks

**Comment by archiewood** [comment]
Sure, how do I update the version?
(Currently using cube cloud)

<img width="539" alt="image" src="https://user-images.githubusercontent.com/58074498/184366169-7d5e2078-e5a0-45be-b5a8-9175e1c24bc7.png">


Do I need to install locally

**Comment by paveltiunov** [comment]
@archiewood You can try right now as the stable version was updated in Cube Cloud.

**Comment by archiewood** [comment]
Thanks.
<img width="492" alt="image" src="https://user-images.githubusercontent.com/58074498/184667915-5494edde-1c23-4185-b759-1ae636acea63.png">

Yes, I just tried it and received the same error. 

One thing that confuses me is that my running query turns up in the Cubecloud query logs, but the one that fails does not:

My two queries:
<img width="283" alt="image" src="https://user-images.githubusercontent.com/58074498/184668719-ebd0b78a-03c6-451f-8204-4437dd61a409.png">

The logs: (this is the same first query 3 times)
<img width="1371" alt="image" src="https://user-images.githubusercontent.com/58074498/184668661-21b29a17-01a4-4baf-af6f-23c8c5a18f41.png">

But it must be getting through as I am getting the error message: 
`SQLCompilationError: Internal: Initial planning error: Error during planning: Table or CTE with name 'not_a_real_table' not found`

**Comment by ovr** [comment]
> SQLCompilationError: Internal: Initial planning error: Error during planning: Table or CTE with name 'not_a_real_table' not found

There is no cube or table from the information schema called `not_a_real_table`.

Probably, we should return specific PostgreSQL code for this case, but it's not a high-priority task. Do I miss something?

Thanks

**Comment by archiewood** [comment]
thanks for the quick followup @ovr 

Yes - I purposely broke the query for example purposes.
The same thing will happen if I make any SQL syntax error (which is a pretty common occurrence when developing SQL how picky the language is about `,` etc)

I guess the point is that I would expect some fail state to be returned so I can handle it in my application, and prompt the user to fix their query, rather than crashing.

For example, when I use a postgres table hosted on Heroku, it won't crash when I make a sql mistake
![CleanShot 2022-08-15 at 12 54 02](https://user-images.githubusercontent.com/58074498/184680117-e53345cc-888e-4c32-9d07-cb0b88b2bcd1.gif)

Compare with the GIF above for Cube.js, where my server crashes

**Comment by ovr** [comment]
@archiewood 

After a short review, I've found what is going wrong. It's a Cube Cloud bug. I've created an issue, it will be fixed in the near time (I cannot provide ETA, but I will write down a message in this issue).

The bug is caused by the Terminate message, which the client sends to the release connection. Cube Cloud has a SQL router that works as ingress to proxy traffic to specific deployment.

Thanks for bug reporting!
